File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        
        ListNode res = new ListNode(0,head);
        ListNode current = res;
        
        while(head != null){
            
            if (head.next != null) {
                ListNode nextNode = head.next;
                if(head.val == nextNode.val){
                    head.next = nextNode.next;
                    continue;
                }
                
            }
            
            head = head.next;
        }
        
        return res.next;
    }
}

File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        
        ListNode* res;
        res = head;
        
        while(head != nullptr){
            
            if(head->next != nullptr){
                ListNode* nextNode = head->next;
                if(head->val == nextNode->val){
                    head->next = nextNode->next;
                    continue;
                }
            }
            
            head = head->next;
            
        }
        
        return res;
    }
};